Constructor-io/connector-akeneo
# [PARTI-67] Support new changes from uuid field

# PR Description

This PR adds the following changes:

- Fetch products by `uuid` instead of `identifier`(sku)
- Made the product's id mapping to fallback to `uuid` when no `sku` field is present
- Changed `npm run dev` command to also run if there are any typescript errors
- Enabled `sourceMap` in `tsconfig.json` for better debugging experience

## ⚠️ Dependencies
- None

## 🕹 Demo

UUID been used as product id
![image](https://user-images.githubusercontent.com/29166076/195435726-f5288b1c-c728-4f6e-b2ee-463e788de98c.png)

## 📝 How to test?

1. Run the connector with your env vars using `npm run dev`
2. Check for ingested products in the [customer interface](https://app.constructor.io/dashboard)
3. Check for ingestion errors in the [catalog upload page](https://app.constructor.io/dashboard/integration/ingestions)

## 🧪 What should be tested?

- The number of products ingested versus the `master` branch, it should be the same number
- If the constructor product id is indeed a UUID field, when there's no `sku` available
  Could also validate if the product matches akeneo's data using this cURL
  ```cURL
  curl --location --request GET 'https://constructorio.demo.cloud.akeneo.com/api/rest/v1/products-uuid/{UUID been searched}' --header 'Content-Type: application/json' --header 'Authorization: Bearer your_token' --header 'Cookie;'
  ```
